momaland/envs/gem_mining/gem_mining.py

```diff
@@ -0,0 +1,274 @@
+    return env
+
+
+def raw_env(**kwargs):
+    """Env factory function for the gem mining domain."""
+    return MOGemMining(**kwargs)
+
+
+class MOGemMining(MOParallelEnv):
+    """Environment for MO gem mining domain.
+
+    The init method takes in environment arguments and should define the following attributes:
+    - possible_agents
+    - action_spaces
+    - observation_spaces
+    These attributes should not be changed after initialization.
+    """
+
+    metadata = {"render_modes": ["human"], "name": "mogemmining_v0"}
+
+    def __init__(
+        self,
+        problem_name="Gem_Mining",
+        num_agents=20,
+        num_objectives=2,
+        min_connectivity=2,
+        max_connectivity=4,
+        min_workers=1,
+        max_workers=5,
+        min_prob=0.01,
+        max_prob=0.50,
+        trunc_probability=0.9,
+        w_bonus=1.03,
+        correlated_objectives=True,
+        num_timesteps=1,
+        render_mode=None,
+        seed=42,
+    ):
+        """Initializes the congestion game.
+
+        Args:
+            problem_name: the name of the network that will be used
+            num_agents: number of agents (i.e., villages) in the Gem Mining instance
+            num_objectives: number of objectives (i.e., gem types), each mine has a probability of generating gems of any type at any timesteps
+            min_connectivity: the minimum number of mines each agent is connected to. Should be greater or equal to 2
+            max_connectivity: the maximum number of mines each agent is connected to. Should be greater or equal to min_connectivity
+            min_workers: the minimum number of workers per village (agent). Should be greater or equal to 1.
+            max_workers: the maximum number of workers per village (agent). Should be greater or equal to min_workers.
+            min_prob: the minimum (Bernoulli) probability of finding a gem (per type) at a mine, excluding worker bonus
+            max_prob: the maximum (Bernoulli) probability of finding a gem (per type) at a mine, excluding worker bonus
+            trunc_probability: upper limit to the probability of finding a gem after adding the worker bonus
+            w_bonus: worker bonus; the probability of finding a gem is multiplied by w_bonus^(w-1), where w is the number of workers at a mine
+            correlated_objectives: if true, the probability of mining a given type of gem at a mine is negatively correlated to finding a gem of another type, and the (non-bonus) expectation of finding any gem is at most max_prob per mine per timestep.
+            num_timesteps: number of timesteps (stateless, therefore always 1 timestep)
+            render_mode: render mode
+            seed: This environment is generated randomly using the provided seed. Defaults to 42.
+        """
+        self.num_timesteps = num_timesteps
+        self.episode_num = 0
+        self.render_mode = render_mode
+
+        self.possible_agents = [i for i in range(num_agents)]
```

Comment:
Does this work? It's supposed to be a list of strings, of the form: "agent_ID". Please update to align with the rest of the environments and PettingZoo.